### PR TITLE
[PM-1071] Display import-details-dialog on successful import

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Instead of relying on an error flow to request a password, the functionality is now provided by a mandatory callback which the `import.component` needs to provide to the `import.service`. This enables all/future importers to easily request a password by simply calling the callback.

On successful import, display a dialog, detailing the total amount of items imported and into which types these were imported.

## Code changes

**Tech Debt:**
- Prefer callback over error-flow to prompt for password (777f9c9)
- Fix up test cases (652c6d1)

**Feature work:**
- In preparation for the other dialogs (success, error)
  - Move file-password-prompt into dialog-folder (149dc0a)
- Add import success dialog (705e3c6)

## Screenshots
image

Requesting a review, by @bitwarden/dept-design to check on the UI, but also the used language.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by jtouchstonebw** [comment]
@djsmith85 Do we need to show anything for item types that the user didn't attempt to import?

**Comment by djsmith85** [comment]
As requested by @danielleflinn, here's a quick video showing the import and import-success dialog:

import-success

@jtouchstonebw: Not sure if I follow. Everything that is present in the source file will be imported, so far the selected importer supports importing an item type.

**Comment by danielleflinn** [comment]
@djsmith85 thanks for adding the recording! I agree with @jtouchstoneb. It would be beneficial to hide any item types from the table that have "0" imported (so Card, Secure Note, Identity, and Collection in your example). That way the user has less to review when looking at this confirmation. This will also keep the modal tidier when we expand to additional or custom item types.

**Comment by djsmith85** [comment]
@danielleflinn, @jtouchstonebw Thank you for the feedback. With the same import example as above the dialog would look like this:

image

File: apps/web/src/locales/en/messages.json [hunk 1]
```diff
@@ -1256,6 +1256,15 @@
   "importSuccess": {
     "message": "Data successfully imported"
   },
+  "importSuccessNumberOfItems": {
+    "message": "We were able to import a total of $AMOUNT$ items.",
```
Comment (danielleflinn) [inline apps/web/src/locales/en/messages.json:1260]: "A total of $AMOUNT$ items were imported."

**Review by danielleflinn — CHANGES_REQUESTED** [review]
This looks good to me, with a minor copy tweak

**Review by danielleflinn — APPROVED** [review]

**Review by aj-rosado — DISMISSED** [review]
Looking great!

File: libs/common/src/importers/importer.ts [hunk 2]
```diff
@@ -2,5 +2,6 @@ import { ImportResult } from "../models/domain/import-result";
 
 export interface Importer {
   organizationId: string;
+  promptForPassword_callback: () => Promise<string>;
```
Comment (MGibson1) [inline libs/common/src/importers/importer.ts:5]: I don't like adding this to an interface implemented by every importer for exclusively bitwarden ones.

Why not just add this to the password protected constructor only?

**Review by aj-rosado — DISMISSED** [review]

**Review by MGibson1 — COMMENTED** [review]
Nothing big, just a few questions

File: libs/importer/src/services/import.service.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ import {
   AvastJsonImporter,
   AviraCsvImporter,
   BitwardenCsvImporter,
-  BitwardenJsonImporter,
+  // BitwardenJsonImporter,
```
[the remark this answers is not in the document]
Comment (djsmith85) [reply libs/importer/src/services/import.service.ts]: Nope, no good reason

**Review by MGibson1 — APPROVED** [review]

**Review by aj-rosado — DISMISSED** [review]

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**